common/tests/unittests/test_deprecated.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+
+from autogluon.common.utils.deprecated import Deprecated, Deprecated_args
+
+
+@Deprecated(min_version_to_warn="0.0", min_version_to_error="9999.0")
+class ShouldDeprecateButNoErrorClass:
+    pass
```

Comment:
Missing test on micro-version changes (even though we probably won't use micro versions for deprecations in general, good to test that it does work correctly)

Reply:
Also: Missing test on pre-release version (Ensure that the logic works correctly in our pre-release version scheme)

Reply:
Sure. Though the version comparison just uses `packaging.version.parse`. So if we assume this package is doing the right thing, there's no need to test that. I can still add a test as a safe guard

Reply:
"So if we assume this package is doing the right thing" : We should not make such assumptions, otherwise why test any of it at all if we assume everything is doing the right thing?